orttraining/orttraining/test/python/orttraining_test_onnxblock.py

```diff
@@ -0,0 +1,517 @@
+        dynamic_axes[input_name] = {}
+        for dim_idx in range(len(model_input.shape)):
+            dynamic_axes[input_name].update({dim_idx: f"{input_name}_dim{dim_idx}"})
+
+    for i, model_output in enumerate(model_outputs):
+        output_name = f"output-{i}"
+        output_names.append(output_name)
+        dynamic_axes[output_name] = {}
+        for dim_idx in range(len(model_output.shape)):
+            dynamic_axes[output_name].update({dim_idx: f"{output_name}_dim{dim_idx}"})
+
+    f = io.BytesIO()
+    torch.onnx.export(
+        torch_model,
+        model_inputs,
+        f,
+        input_names=input_names,
+        output_names=output_names,
+        opset_version=14,
+        do_constant_folding=False,
+        training=torch.onnx.TrainingMode.TRAINING,
+        dynamic_axes=dynamic_axes,
+        export_params=True,
+        keep_initializers_as_inputs=False,
+    )
+    return onnx.load_model_from_string(f.getvalue())
+
+
+def _to_numpy(tensor):
+    return (
+        tensor.detach().cpu().numpy() if tensor.requires_grad else tensor.cpu().numpy()
+    )
+
+
+# All unit tests
+
+
+@pytest.mark.parametrize(
+    "graph",
+    [
+        SimpleGraphWithMSELoss,
+        SimpleGraphWithCrossEntropyLoss,
+        SimpleTrainingGraphWithMSELoss,
+        SimpleTrainingGraphWithCrossEntropyLoss,
+    ],
+)
+def test_loss_composition(graph):
+    # Given
+    device = "cuda"
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = SimpleNet(D_in, H, D_out).to(device)
+    x = torch.randn(N, D_in, device=device)
+    onnx_model = _get_onnx_model(model, (x,))
+
+    # When / Then no error occurs
+    simple_graph = graph(onnx_model)
+    lossful_graph = simple_graph()
+
+
+def test_mse_loss_execution():
+    # Given
+    device = "cuda"
+    N, D_in, H, D_out = 64, 784, 500, 10
+    model = SimpleNet(D_in, H, D_out).to(device)
+    x = torch.randn(N, D_in, device=device)
+    target = torch.randn(N, D_out, device=device)
+    onnx_model = _get_onnx_model(model, (copy.deepcopy(x),))
+    simple_graph = SimpleGraphWithMSELoss(onnx_model)
+    lossful_model = simple_graph()
+    model_name = "model_with_mse_loss.onnx"
+    ort_output_names = [lossful_model.graph.output[0].name]
+    ort_inputs = {
+        lossful_model.graph.input[0].name: _to_numpy(copy.deepcopy(x)),
+        lossful_model.graph.input[1].name: _to_numpy(copy.deepcopy(target)),
+    }
+
+    def mse_loss(prediction, target):
+        loss = torch.nn.MSELoss()
+        return loss(prediction, target)
+
+    # When
+    with tempfile.TemporaryDirectory() as onnx_dir_name:
+        onnx_file_path = os.path.join(onnx_dir_name, model_name)
+        onnx.save(lossful_model, onnx_file_path)
+        ort_session = onnxruntime.InferenceSession(
+            onnx_file_path, providers=C.get_available_providers()
+        )
+
+        ort_outs = ort_session.run(ort_output_names, ort_inputs)
+        torch_outs = mse_loss(model(copy.deepcopy(x)), copy.deepcopy(target))
```

Comment:
since you are already copying ip/target for ort run, maybe you can skip for pytorch run